app/models/zipcode.rb

```diff
@@ -1,3 +1,26 @@
 class Zipcode < ActiveRecord::Base
   validates_presence_of :zipcode
+
+  def self.create_zipcode(string)
+    clean = clean_zipcode(string)
+    find_or_create_by(zipcode: clean)
+  end
+
+  def self.clean_zipcode(string)
```

Comment:
This is pretty good, though I think it could be shorter by using [rjust](https://ruby-doc.org/core-2.3.1/String.html#method-i-rjust). Also, we should have a condition to catch nil zipcode values and return them as '00000'